exercises/bowling/bowling_test.rb

```diff
@@ -3,224 +3,200 @@
-      'Pins must have a value from 0 to 10') do
-        @game.roll(-1)
-      end
+    assert_raises StandardError do
+      roll([-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+      @game.score
+    end
   end
 
-  def test_should_not_allow_rolls_better_than_strike
+  def test_a_roll_can_not_score_more_than_10_points
     skip
-    assert_raises(
-      RuntimeError,
-      'Pins must have a value from 0 to 10') do
-        @game.roll(11)
-      end
+    assert_raises StandardError do
+      roll([11, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+      @game.score
+    end
   end
 
-  def test_should_not_allow_two_normal_rolls_better_than_strike
+  def test_two_rolls_in_a_frame_can_not_score_more_than_10_points
     skip
-    assert_raises RuntimeError, 'Pin count exceeds pins on the lane' do
-      @game.roll(5)
-      @game.roll(6)
+    assert_raises StandardError do
+      roll([5, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+      @game.score
     end
   end
 
-  def test_should_not_allow_two_normal_rolls_better_than_strike_in_last_frame
+  def test_two_bonus_rolls_after_a_strike_in_the_last_frame_can_not_score_more_than_10_points
     skip
-    roll_n_times(18, 0)
-    assert_raises RuntimeError, 'Pin count exceeds pins on the lane' do
-      @game.roll(10)
-      @game.roll(5)
-      @game.roll(6)
+    assert_raises StandardError do
+      roll([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 5, 6])
+      @game.score
     end
   end
 
-  def test_should_not_allow_to_take_score_at_the_beginning
+  def test_an_unstarted_game_can_not_be_scored
     skip
-    assert_raises(
-      RuntimeError,
-      'Score cannot be taken until the end of the game',
-    ) do
+    assert_raises StandardError do
+      roll([])
       @game.score
     end
   end
 
-  def test_should_not_allow_to_take_score_before_game_has_ended
+  def test_an_incomplete_game_can_not_be_scored
     skip
-    roll_n_times(19, 5)
-    assert_raises(
-      RuntimeError,
-      'Score cannot be taken until the end of the game') do
-        @game.score
-      end
+    assert_raises StandardError do
+      roll([0, 0])
+      @game.score
+    end
   end
 
-  def test_should_not_allow_rolls_after_the_tenth_frame
+  def test_a_game_with_more_than_ten_frames_can_not_be_scored
     skip
-    roll_n_times(20, 0)
-    assert_raises(
-      RuntimeError,
-      'Should not be able to roll after game is over',
-    ) do
-      @game.roll(0)
+    assert_raises StandardError do
+      roll([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+      @game.score
     end
   end
 
-  def test_should_not_calculate_score_before_fill_balls_have_been_played
+  def test_bonus_rolls_for_a_strike_in_the_last_frame_must_be_rolled_before_score_can_be_calculated
     skip
-    roll_n_times(10, 10)
+    assert_raises StandardError do
+      roll([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10])
+      @game.score
```

Comment:
Some of the test cases expect an exception to be raised by `game.roll` and an invalid roll while some are raised by `game.score` when the game is incomplete. 

As there is no data in x-common to indicate whether the exception could be raised by `score` or by `roll`, it is difficult to procedurally generate test cases to include or exclude `game.score` from  test cases. I'm open to any ideas.

The x-common data for exceptions sets the expected value as -1 and the decision of when to error is left up to the implementation (error when an invalid roll is made, error when scoring a game with an invalid roll, or error in both cases). With the current test cases, an implemented solution may decide to only error on `game.score`, `game.roll`, or both. This way a person could implement a solution that never errors on `roll` but only error on `score`.